components/powersensor/ina/ina.go

```diff
@@ -277,7 +280,14 @@ func (d *ina) Power(ctx context.Context, extra map[string]interface{}) (float64,
 	}
 	defer utils.UncheckedErrorFunc(handle.Close)
 
-	pow, err := handle.ReadRegU16BE(powerRegister)
+	// Calibrate each time the power value is read, so if anything else is also writing to these registers
+	// we have the correct value.
+	err = d.calibrate()
+	if err != nil {
+		return 0, err
+	}
+
+	pow, err := handle.ReadRegS16BE(powerRegister)
```

Comment:
There's a lot of duplicated code in here. Mild preference for refactoring into a helper function, but I don't feel strongly.
```go
func (d *ina) readSensor(register int, lsb int) (float64, error) {
	handle, err := i2c.NewI2C(d.addr, d.bus)
	if err != nil {
		d.logger.Errorf("can't open ina i2c handle: %s", err)
		return 0, err
	}
	defer utils.UncheckedErrorFunc(handle.Close)

	// Calibrate each time the power value is read, so if anything else is also writing to these registers
	// we have the correct value.
	err = d.calibrate()
	if err != nil {
		return 0, err
	}

	rawValue, err := handle.ReadRegS16BE(register)

	if err != nil {
		return 0, err
	}
	value := fromNano(float64(int64(rawValue) * lsb))
	return value, nil
}

func (d *ina) Power(ctx context.Context, extra map[string]interface{}) (float64, error) {
	return d.readSensor(powerRegister, d.powerLSB)
}

func (d *ina) Current(ctx context.Context, extra map[string]interface{}) (float64, error) {
	return d.readSensor(currentRegister, d.currentLSB)
}
```